### PR TITLE
Api test fixtures

### PR DESCRIPTION
**Pull request by Arminek**
Added:
Doctrine configuration
Console component
Serialize component

### CODE REVIEW

File: src/ApiTestCase.php [hunk 1]
```diff
@@ -154,6 +170,33 @@ protected function getJsonResponseFixture($filename)
...
+    private function persistObjects($objects)
+    {
+        foreach ($objects as $object) {
+            $this->entityManager->persist($object);
+        }
+    }
+
+    private function getFixturesFolder()
+    {
+        return $this->getRootDir().'/../src/DataFixtures/ORM/';
```
Comment (lchrusciel) [inline src/ApiTestCase.php]: We should make it configurable with `/../src/DataFixtures/ORM/` as a default value.

File: src/XmlApiTestCase.php [hunk 2]
```diff
@@ -17,13 +17,13 @@
  * @author Łukasz Chruściel <lukasz.chrusciel@lakion.com>
  * @author Paweł Jędrzęjewski <pawel.jedrzejewski@lakion.com>
  * @author Michał Marcinkowski <michal.marcinkowski@lakion.com>
- * @author Arkadiusz Krakowiak <arkadiusz.krakowiak@lakion.com>
```
Comment (lchrusciel) [inline src/XmlApiTestCase.php]: Don't be so shy ;) You introduced this files.

File: test/app/AppKernel.php [hunk 3]
```diff
@@ -41,6 +42,27 @@ public function registerContainerConfiguration(LoaderInterface $loader)
                     'resource' => '%kernel.root_dir%/config/routing.yml',
                 ),
             ));
+            $container->loadFromExtension('doctrine', array(
+                'dbal' => array(
+                    'driver' => 'pdo_mysql',
+                    'dbname' => 'api_test_case',
+                    'host' => 'localhost',
+                    'user' => 'root',
+                    'password' => 'toor',
```
Comment (lchrusciel) [inline test/app/AppKernel.php]: Never commit your local information! We should extract this to some external file with default values.

File: test/app/config/doctrine/product.orm.yml [hunk 4]
```diff
@@ -0,0 +1,14 @@
+Lakion\ApiTestCase\Test\Entity\Product:
+    type: entity
+    table: api_test_product
```
Comment (lchrusciel) [inline test/app/config/doctrine/product.orm.yml]: I think `app_product` or `test_product` will be enough.

File: test/app/config/doctrine/product.orm.yml [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+        id:
+            type: integer
+            id: true
+            generator:
+                strategy: AUTO
+    fields:
+        name:
+            type: string
+        price:
+            type: integer
```
Comment (lchrusciel) [inline test/app/config/doctrine/product.orm.yml]: Missing new line.

File: composer.json [hunk 6]
```diff
@@ -24,7 +24,13 @@
...
         "symfony/framework-bundle": "^2.3|^3.0",
         "symfony/finder": "^2.3|^3.0",
-        "symfony/browser-kit": "^2.3|^3.0"
+        "symfony/browser-kit": "^2.3|^3.0",
+        "nelmio/alice": "@dev",
+        "doctrine/orm": "^2.5",
+        "doctrine/data-fixtures": "^1.1",
+        "doctrine/doctrine-bundle": "^1.6",
+        "symfony/console": "^3.0",
+        "symfony/serializer": "^3.0"
```
Comment (pjedrzejewski) [inline composer.json]: Shouldn't it be same as for browser-kit?

Comment (lchrusciel) [inline composer.json]: :+1:

File: .travis.yml [hunk 7]
```diff
@@ -22,6 +22,9 @@ env:
 
 before_script:
   - composer update --prefer-source PREFER_LOWEST
```
Comment (pamil) [inline .travis.yml]: `$PREFER_LOWEST`

File: test/app/config/parameters.php.dist [hunk 8]
```diff
@@ -0,0 +1,13 @@
...
+ * This file is part of the ApiTestCase package.
+ *
+ * (c) Lakion
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+$container->setParameter('database.user', 'root');
+$container->setParameter('database.password', '');
```
Comment (lchrusciel) [inline test/app/config/parameters.php.dist]: Missing new line.

**Pull request merged**